public/app/features/provisioning/RepositoryListPage.tsx

```diff
@@ -51,36 +64,73 @@ function RepositoryListPageContent({ items }: { items?: Repository[] }) {
         {!!filteredItems.length ? (
           filteredItems.map((item) => {
             const name = item.metadata?.name ?? '';
-            const healthy = Boolean(item.status?.health.healthy)
+            const healthy = Boolean(item.status?.health.healthy);
+            let icon: IconName = 'database'; // based on type
+            let meta: ReactNode[] = [
+              // TODO... add counts? and sync info
+            ];
+            switch (item.spec?.type) {
+              case 'github':
+                icon = 'github';
+                const spec = item.spec.github;
+                let url = `https://github.com/${spec?.owner}/${spec?.repository}/`;
+                if (spec?.branch) {
+                  url += `tree/` + spec?.branch;
+                }
+                meta.push(<a href={url}>{url}</a>);
+                break;
+
+              case 'local':
+                meta.push(item.spec.local?.path);
+                break;
+            }
             return (
-              <Card key={item.metadata?.name} className={css({ alignItems: 'center' })}>
-                <Stack direction={'column'}>
-                  <Card.Heading>
-                    <TextLink href={`${PROVISIONING_URL}/${name}`}>{item.spec?.title}</TextLink>
-                  </Card.Heading>
-                  <Card.Meta>{item.spec?.type}</Card.Meta>
-                </Stack>
-                <Stack>
-                  {healthy ? <>
-                    <LinkButton variant="secondary" href={`${PROVISIONING_URL}/${name}/edit`}>
-                      Edit
+              <Card key={item.metadata?.name}>
+                <Card.Figure>
+                  <Icon name={icon} width={40} height={40} />
+                </Card.Figure>
+                <Card.Heading>{item.spec?.title}</Card.Heading>
+                <Card.Description>
+                  {item.spec?.description}
+
+                  {item.status ? (
+                    <>
+                      {!healthy && (
+                        <Alert
+                          title="Repository is unhealthy"
+                          children={item.status?.health?.message?.map((v) => (
+                            <div>
+                              {v}
+                              <br />
+                              <br />
+                            </div>
+                          ))}
+                        ></Alert>
+                      )}
+                    </>
+                  ) : (
+                    <div>
+                      <Alert severity="warning" title="repository initializing" />
```

Comment:
for the couple seconds after creating a repository, it will not yet have a status

Reply:
ideally this page is watching the repositorys and keeping them up-to-date, so would re-render on change